docs/lang/attributes.md

```diff
@@ -51,6 +51,15 @@ The entrypoint for the Calyx program. If no component has this attribute, then
 the compiler looks for a component named `main`. If neither is found, the
 compiler errors out.
 
+### `go`, `done`, `clk` and `reset`
+These four ports are part of the interface to Calyx components.
+`go` and `done` provide the mechanism for how an "outer" component invokes an "inner" cell that it contains.
+`clk` and `reset` thread through the global clock and resetting signal in a design.
+
+### `nointerface`
+By default, interface ports are automatically added to a component if they are missing.
+Adding this attribute disables this behavior.
```

Comment:
Could be worth pointing out _who_ adds interface ports—namely, this happens during parsing (not, like, a special lowering pass).

Reply:
These are roundtripping tests so the input file is the same as the output file